### PR TITLE
Let the terminal report through when --report-json names a file

### LINKED ISSUE #145: [Feature] --report-json should be able to coexist with the terminal report
## Summary

Let a run write its JSON report to a file and still print the human-readable report to the terminal, instead of having to choose one.

## Motivation

`--report-json` disables the logger for the whole run, in `src/options.cpp`:

```cpp
if (is_set(program_args::arg::report_json))
{
  cuke::log::disable();
}
```

That is right when the report goes to stdout, where a log line would sit inside the JSON and corrupt it. When a file is named, the JSON is not on stdout at all and the terminal is free, but it stays silent anyway.

The effect today, on the same feature file with one passing step:

```
$ ./cucumber f.feature
Feature: a feature  f.feature:1

Scenario: a scenario  f.feature:3
[   PASSED    ] Given a step  f.feature:4

1 Scenario (1 passed)
1 Step (1 passed)

$ ./cucumber f.feature --report-json r.json
$ wc -c < r.json
714
```

The second run prints nothing at all. Not less, nothing.

The use case is a build system that wants both: the report for a later step to read, and the progress on the terminal so that a long suite is not a silent wait. Right now that needs either two runs of the same suite or a script thatparses the JSON to re-print what the runner already knew how to print.

## Proposed API / Behavior

`--report-json FILE` keeps the terminal output, and a new flag turns it off for anyone who wants today's behaviour:

| Invocation | Terminal | JSON |
| --- | --- | --- |
| no flags | run and summary | none |
| `--report-json` | nothing | stdout |
| `--report-json out.json` | run and summary | `out.json` |
| `--report-json out.json --disable-user-report` | nothing | `out.json` |
| `--disable-user-report` | nothing | none |

`--disable-user-report` is off by default and takes no value. `--report-json` with no file keeps silencing the terminal whether or not the flag is present, because protecting stdout is not a preference.

**Feature file:**
```gherkin
Feature: a feature

  Scenario: a scenario
    Given a step
```

**Step definition:**
```cpp
GIVEN(a_step, "a step")
{
}
```

```sh
# both
./cucumber f.feature --report-json out.json

# today's behaviour, explicitly
./cucumber f.feature --report-json out.json --disable-user-report
```

**This changes the behaviour of `--report-json` for existing callers.** Anyone passing `--report-json out.json` and relying on an empty stdout will start seeing the run on it. I think the new default is the better one, because the silence was a side effect of protecting stdout and it applied in the case where stdout was not carrying the report. If you would rather not move the default, the same flag inverted gives the identical matrix and I am happy to send it that way instead.

## Alternatives Considered

`--quiet` does not help. It sets a log level, and `--report-json` calls `log::disable()`, which is independent of level, so the output is gone whatever the level says. `--quiet` also keeps the final summary, which is a different thing from what is wanted here.

Running the suite twice doubles the time and can disagree with itself.

Having the caller pretty-print the JSON means reimplementing output the runner already produces.

## Additional Context

`write_to_file_or_stdout` in `src/util.hpp` already makes exactly the distinction the logging decision needs, by testing whether the path is empty, so the two places can be made to agree.

I have this implemented and tested and am happy to open it as a pull request.

### PR DESCRIPTION
**Pull request by pinhopro**
## Summary

Enable the terminal output when the JSON report goes to a file sink, aligning with cucumber-js.

## Motivation

Closes #145

`--report-json` disabled the logger for the whole run. That protects stdout when the report goes there, and it is not needed when a file is named: the JSON is not on stdout and the terminal is free, but it stayed silent anyway. A build could watch the suite or keep its report, never both.

## Changes

Two conditions, no new option:

- `src/options.cpp` silences the logger only when the report goes to stdout, the same test `write_to_file_or_stdout` already makes on the path being empty.
- `src/cucumber.cpp` prints the final summary alongside the JSON when a file is named, instead of choosing between them.

The result is the two lines from the review:

- `--report-json` writes JSON to stdout and the terminal stays silent, because the summary would sit inside the JSON.
- `--report-json out.json` writes the JSON there and the terminal behaves as usual.

An earlier revision of this branch added a `--disable-user-report` flag to restore the old silence. It is gone: `--report-json` alone is already the JSON-only mode, and a second suppression option beside `-q`/`--quiet` is one the reader has to disambiguate by experiment.

**This changes what an existing caller sees.** Anyone passing `--report-json out.json` and relying on an empty stdout will start seeing the run on it.

## How to Test

```sh
cmake --build ./build -j$(nproc)

# Unit tests
./build/bin/unittests

# Integration tests
./build/bin/stress-tests ./stress-tests
```

591 unit tests pass, 33 stress scenarios pass, example smoke unchanged. Rebased onto `main` after #144 merged.

By hand, on a bundled example:

```sh
./build/bin/example ./examples/features/1_first_scenario.feature
#   2 Scenarios (2 passed), 8 Steps (8 passed)

./build/bin/example ./examples/features/1_first_scenario.feature --report-json
#   the JSON, and nothing else

./build/bin/example ./examples/features/1_first_scenario.feature --report-json /tmp/r.json
#   the same terminal output as the first, and /tmp/r.json written
```

New tests in `gtest/options.cc` cover all three cases, at the logger level and through `print_results()`.

The scanner, lexer, parser and step-matching code are untouched, so no fuzz run applies.

## Checklist

- [x] Code follows the Allman brace style and passes `clang-format`
- [x] New or changed behaviour is covered by a unit test in `gtest/` or a scenario in `stress-tests/`
- [x] No mandatory new dependencies introduced
- [x] `AGENTS.md` updated if new APIs, conventions, or CLI flags were added

## Summary by CodeRabbit

* **Bug Fixes**
  * Specifying a file path with `--report-json` now preserves normal terminal logging and the scenario summary while writing the JSON report to the file.
  * Using bare `--report-json` continues to suppress terminal output during the run and writes the JSON report to standard output.

### CODE REVIEW

File: src/cucumber.cpp [hunk 1]
```diff
@@ -67,18 +67,23 @@ void cwt_cucumber::run_tests() const noexcept
...
+  // attempted first; --quiet suppresses it (including this function's own
+  // log::report() calls) by having already disabled the logger in
+  // program_args::initialize(). The JSON is then written whenever
+  // --report-json is set, in addition to whatever the human report above
+  // printed, which is why a bare --report-json shows the run and the
+  // summary before the JSON that follows it on stdout.
+  print_failed_scenarios();
+  log::report(log::new_line);
+  log::report(results::scenarios_to_string(), log::new_line);
+  log::report(results::steps_to_string(), log::new_line);
```
Comment (ThoSe1990) [inline src/cucumber.cpp]: Lets control the summary prints with a `--no-summary` argument

File: src/log.hpp [hunk 2]
```diff
@@ -27,7 +27,6 @@ enum class level
 {
   verbose = 0,
   info,
-  quiet,
```
Comment (ThoSe1990) [inline src/log.hpp]: lets keep the log level `quiet`

File: src/options.hpp [hunk 3]
```diff
@@ -125,7 +125,7 @@ class program_args
   // clang-format off
   static constexpr const definition defs[] = {
       {"-h", "--help", "", arg::help, arg_type::flag, "Print the help screen to stdout"},
-      {"-q", "--quiet", "", arg::quiet, arg_type::flag, "Quiet mode, only the final result will be printed to stdout."},
+      {"-q", "--quiet", "", arg::quiet, arg_type::flag, "Quiet mode, suppress all terminal output, including the final summary."},
```
Comment (ThoSe1990) [inline src/options.hpp]: old description + `--no-summary`

File: CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,7 @@
 ### Changed
 
 - Step matching now reuses a precompiled regex per step definition instead of recompiling it on every match attempt ([136](https://github.com/ThoSe1990/cwt-cucumber/pull/136))
+- `--quiet` now suppresses the human report on the terminal entirely, including the final summary and a failed assertion's message; before, the summary still printed ([145](https://github.com/ThoSe1990/cwt-cucumber/issues/145))
```
Comment (ThoSe1990) [inline CHANGELOG.md]: no change with `--quiet`

File: docs/chapters/program_arguments.rst [hunk 5]
```diff
@@ -69,7 +71,10 @@ all steps will run regardless of intermediate failures.
 JSON Report: ``--report-json``
 ------------------------------
 
-Print test results as JSON to stdout or a specified file.  
+Print test results as JSON: to stdout with no argument, to the named file
+with one. It says nothing about the human report, which prints as usual
+either way (unless ``--quiet`` suppresses it); with no file named, the
+JSON follows the human report on stdout.
```
Comment (ThoSe1990) [inline docs/chapters/program_arguments.rst]: the documentation will change accordingly
